extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,320 @@
+    The device pointer can only be used for arguments that are declared to be a
+    pointer to `global` memory allocated with clCreateBuffer() with the
+    CL_MEM_DEVICE_SHARED_ADDRESS_EXT flag. The pointer value specified as the argument value
+    can be the pointer to the beginning of the buffer or be a pointer offset into
+    the buffer region. The device pointer value must be naturally aligned according to
+    the argument's type.
+
+// refError
+
+{clSetKernelArgDevicePointerEXT} returns {CL_SUCCESS} if the function was executed
+successfully. Otherwise, it returns one of the following errors:
+
+  * {CL_INVALID_KERNEL} if _kernel_ is not a valid kernel object.
+  * {CL_INVALID_OPERATION} if no devices in the context associated with _kernel_ support
+    the device pointer.
+  * {CL_INVALID_ARG_INDEX} if _arg_index_ is not a valid argument index.
+  * {CL_INVALID_ARG_VALUE} if _arg_value_ specified is not a valid value.
+  * {CL_OUT_OF_RESOURCES} if there is a failure to allocate resources required
+    by the OpenCL implementation on the device.
+  * {CL_OUT_OF_HOST_MEMORY} if there is a failure to allocate resources
+    required by the OpenCL implementation on the host.
+
+Add a new flag to clSetKernelExecInfo for setting indirect device pointer access info <<clSetKernelExecInfo, List of supported param_name stable>>:
+
+[width="100%",cols="<33%,<17%,<50%",options="header"]
+|====
+| Kernel Exec Info | Type | Description
+| {CL_KERNEL_EXEC_INFO_DEVICE_PTRS_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_KERNEL_EXEC_INFO_DEVICE_PTRS_EXT.asciidoc[]
+  | {cl_mem_device_address_EXT_TYPE}
+      | Device pointers must reference locations contained entirely within
+        buffers that are passed to kernel as arguments, or that are passed
+        through the execution information.
+
+        Non-argument device pointers accessed by the kernel must be specified
+        by passing pointers to those buffers via {clSetKernelExecInfo}.
+|====
+
+// refError
+
+Change the descriptions for when returning CL_INVALID_OPERATION from {clSetKernelExecInfo}
+as follows:
+
+  * {CL_INVALID_OPERATION} if passing {CL_KERNEL_EXEC_INFO_SVM_PTRS} or
+  {CL_KERNEL_EXEC_INFO_SVM_FINE_GRAIN_SYSTEM} with _param_value_ set to CL_TRUE
+  and no device in the context associated with _kernel_ support SVM.
+  * {CL_INVALID_OPERATION} if passing {CL_KERNEL_EXEC_INFO_DEVICE_PTRS_EXT} and no
+  device in the context associated with _kernel_ support device pointers.
+
+== Interactions with Other Extensions
+
+This extension is targeted to complement the OpenCL SVM extension and/or the
+Intel Unified Shared Memory extension by providing an additional lower-end
+step in the spectrum of type of pointers/buffers OpenCL can allocate. The
+extension can be seen as a simplification of the USM Device allocation type
+which drops the need to map the device buffer's address range to the same
+position in the host memory or to implement platform-wide VM.
```

Comment:
Strictly speaking this is not really an interaction. Consider folding this (useful) text into the (already very helpful) overview.